src/group_theory/group_action/prod.lean

```diff
@@ -45,6 +46,17 @@ instance has_faithful_scalar_right [nonempty α] [has_faithful_scalar M β] :
 
 end
 
+@[to_additive]
+instance smul_comm_class_both [monoid N] [monoid P] [has_scalar M N] [has_scalar M P]
+  [smul_comm_class M N N] [smul_comm_class M P P] :
+  smul_comm_class M (N × P) (N × P) :=
+⟨λ c x y, by simp [smul_def, mul_def, mul_smul_comm]⟩
+
+instance is_scalar_tower_both [monoid N] [monoid P] [has_scalar M N] [has_scalar M P]
```

Comment:
```suggestion
instance is_scalar_tower [monoid N] [monoid P] [has_scalar M N] [has_scalar M P]
```
to match `pi.is_scalar_tower`

Reply:
This is a match to `pi.is_scalar_tower'`, not `pi.is_scalar_tower`, and I prefer to use something more meaningful than `'`.